### PR TITLE
[FLINK-18384][docs] translate "Elasticsearch SQL Connector" page into Chinese

### PR DESCRIPTION
**Pull request by jjiey**
## What is the purpose of the change

[FLINK-18384](https://issues.apache.org/jira/browse/FLINK-18384)
Translate "Elasticsearch SQL Connector" page into Chinese.

## Brief change log

* Translate "Elasticsearch SQL Connector" page into Chinese, the doc file is located in 'flink/docs/content.zh/docs/connectors/table/elasticsearch.md'.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by jjiey** [comment]
Hi, @wuchong, could you help to review this? Thanks.

**Comment by wuchong** [comment]
cc @leonardBang , could you help to review this?

**Comment by leonardBang** [comment]
> cc @leonardBang , could you help to review this?

ok, I'll take a look

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 1]
```diff
@@ -24,26 +24,26 @@ specific language governing permissions and limitations
...
 -->
 
-# Elasticsearch SQL Connector
+# Elasticsearch SQL 连接器
 
 {{< label "Sink: Batch" >}}
 {{< label "Sink: Streaming Append & Upsert Mode" >}}
 
-The Elasticsearch connector allows for writing into an index of the Elasticsearch engine. This document describes how to setup the Elasticsearch Connector to run SQL queries against Elasticsearch.
+Elasticsearch 连接器允许将数据写入到 Elasticsearch 引擎的索引中。本文档描述如何设置 Elasticsearch 连接器来对 Elasticsearch 运行 SQL 查询。
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:32]:
```suggestion
Elasticsearch 连接器允许将数据写入到 Elasticsearch 引擎的索引中。本文档描述运行 SQL 查询时如何设置 Elasticsearch 连接器。
```

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 2]
```diff
@@ -24,26 +24,26 @@ specific language governing permissions and limitations
...
 
-The connector can operate in upsert mode for exchanging UPDATE/DELETE messages with the external system using the primary key defined on the DDL.
+连接器可以在 upsert 模式下操作，使用 DDL 中定义的主键与外部系统交换 UPDATE/DELETE 消息。
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:34]:
```suggestion
连接器可以工作在 upsert 模式，使用 DDL 中定义的主键与外部系统交换 UPDATE/DELETE 消息。
```

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 3]
```diff
@@ -24,26 +24,26 @@ specific language governing permissions and limitations
...
 
-If no primary key is defined on the DDL, the connector can only operate in append mode for exchanging INSERT only messages with external system.
+如果 DDL 中没有定义主键，那么连接器只能在 append 模式下操作，只能与外部系统交换 INSERT 消息。
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:36]:
```suggestion
如果 DDL 中没有定义主键，那么连接器只能工作在 append 模式，只能与外部系统交换 INSERT 消息。
```

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 4]
```diff
@@ -59,212 +59,208 @@ CREATE TABLE myUserTable (
 );
 ```
 
-Connector Options
+连接器选项
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:62]:
```suggestion
连接器参数
```

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 5]
```diff
@@ -59,212 +59,208 @@ CREATE TABLE myUserTable (
...
 
 <table class="table table-bordered">
     <thead>
       <tr>
-        <th class="text-left" style="width: 25%">Option</th>
-        <th class="text-center" style="width: 8%">Required</th>
-        <th class="text-center" style="width: 7%">Default</th>
-        <th class="text-center" style="width: 10%">Type</th>
-        <th class="text-center" style="width: 50%">Description</th>
+        <th class="text-left" style="width: 25%">选项</th>
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:68]: 参数

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 6]
```diff
@@ -59,212 +59,208 @@ CREATE TABLE myUserTable (
...
+        <th class="text-center" style="width: 8%">是否必选</th>
+        <th class="text-center" style="width: 7%">默认值</th>
+        <th class="text-center" style="width: 10%">类型</th>
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:71]: `数据类型` just keep same with other connector pages.

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 7]
```diff
@@ -59,212 +59,208 @@ CREATE TABLE myUserTable (
...
     <thead>
       <tr>
-        <th class="text-left" style="width: 25%">Option</th>
-        <th class="text-center" style="width: 8%">Required</th>
-        <th class="text-center" style="width: 7%">Default</th>
-        <th class="text-center" style="width: 10%">Type</th>
-        <th class="text-center" style="width: 50%">Description</th>
+        <th class="text-left" style="width: 25%">选项</th>
+        <th class="text-center" style="width: 8%">是否必选</th>
+        <th class="text-center" style="width: 7%">默认值</th>
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:70]: Good translation, I found existed zh docs use `默认参数` which is not properly, could you help update them(upsert-kafka.md, datagen.md, hbase.md, print.md)?

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 8]
```diff
@@ -59,212 +59,208 @@ CREATE TABLE myUserTable (
...
+      <td>最大回退重试次数。</td>
     </tr>
     <tr>
       <td><h5>sink.bulk-flush.backoff.delay</h5></td>
-      <td>optional</td>
+      <td>可选</td>
       <td style="word-wrap: break-word;">50ms</td>
       <td>Duration</td>
-      <td>Delay between each backoff attempt. For <code>CONSTANT</code> backoff, this is simply the delay between each retry. For <code>EXPONENTIAL</code> backoff, this is the initial base delay.</td>
+      <td>每次回退尝试之间的延迟。<code>CONSTANT</code> 回退只是每次重试之间的延迟。<code>EXPONENTIAL</code> 回退是初始基准延迟。</td>
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:204]:
```suggestion
      <td>每次回退尝试之间的延迟。对于<code>CONSTANT</code> 回退策略， 该值是每次重试之间的延迟。对于<code>EXPONENTIAL</code> 回退策略，该值是初始的延迟。</td>
```

File: docs/content.zh/docs/connectors/table/elasticsearch.md [hunk 9]
```diff
@@ -59,212 +59,208 @@ CREATE TABLE myUserTable (
...
+Elasticsearch sink 可以根据是否定义了主键来确定是在 upsert 模式还是 append 模式下工作。
+如果定义了主键，Elasticsearch sink 将以 upsert 模式工作，该模式可以消费包含 UPDATE/DELETE 消息的查询。
+如果未定义主键，Elasticsearch sink 将以 append 模式工作，该模式只能消费包含 INSERT 消息的查询。
 
-In Elasticsearch connector, the primary key is used to calculate the Elasticsearch document id, which is a string of up to 512 bytes. It cannot have whitespaces.
-The Elasticsearch connector generates a document ID string for every row by concatenating all primary key fields in the order defined in the DDL using a key delimiter specified by `document-id.key-delimiter`.
-Certain types are not allowed as primary key field as they do not have a good string representation, e.g. `BYTES`, `ROW`, `ARRAY`, `MAP`, etc.
-If no primary key is specified, Elasticsearch will generate a document id automatically.
+在 Elasticsearch 连接器中，主键用于计算 Elasticsearch 的文档 id，文档 id 为最多 512 字节且不包含空格的字符串。
+Elasticsearch 连接器通过使用 `document-id.key-delimiter` 指定的键分隔符按照 DDL 中定义的顺序连接所有主键字段，为每一行生成一个文档 ID 字符串。
```
Comment (leonardBang) [inline docs/content.zh/docs/connectors/table/elasticsearch.md:242]: 每一行记录

**Review by leonardBang — COMMENTED** [review]
Thanks @jjiey for the contribution, the pr really well, I left some minor comments

File: docs/content.zh/docs/connectors/table/elasticsearch.md (hunk 7 above)
Comment (jjiey) [reply docs/content.zh/docs/connectors/table/elasticsearch.md:70]: ok, I will update them.

**Comment by jjiey** [comment]
Thanks @leonardBang for your review comments. I've updated and commited the modifications.

**Review by leonardBang — APPROVED** [review]
LGTM +1

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**